File: python/7 kyu/133.py
# https://www.codewars.com/kata/5a2d70a6f28b821ab4000004/train/python

# Passed

def largest_sum(s):
    numbers = [int(num) for num in s.split("0") if num]
    digit_sums = [0] # incase s is "0"
    for num in numbers:
        digits = [int(digit) for digit in str(num)]
        digit_sums.append(sum(digits))

    result = max(digit_sums)
    return result

output = largest_sum("72102450111111090")
print(output)

File: nodejs/7 kyu/133.js
// https://www.codewars.com/kata/5574940eae1cf7d520000076/train/javascript

// Passed

function pattern(n) {
  if (n <= 0) return "";
  let numberStrings = [];
  let nearestOddNumber = n % 2 === 0 ? n - 1 : n;
  for (let num = 1; num <= nearestOddNumber; num += 2) {
    let numberString = String(String(num).repeat(num));
    numberStrings.push(numberString);
  }

  let result = numberStrings.join("\n");
  return result;
}

const output = pattern(4);
console.log(output);